File: app/src/main/java/org/michaelbel/moviemade/ui/modules/keywords/KeywordRepository.java
package org.michaelbel.moviemade.ui.modules.keywords;

import org.jetbrains.annotations.NotNull;
import org.michaelbel.moviemade.BuildConfig;
import org.michaelbel.moviemade.Moviemade;
import org.michaelbel.moviemade.data.entity.MoviesResponse;
import org.michaelbel.moviemade.data.service.KeywordsService;
import org.michaelbel.moviemade.utils.AdultUtil;
import org.michaelbel.moviemade.utils.TmdbConfigKt;

import io.reactivex.Observable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class KeywordRepository implements KeywordContract.Repository {

    private KeywordsService service;

    KeywordRepository(KeywordsService service) {
        this.service = service;
    }

    @NotNull
    @Override
    public Observable<MoviesResponse> getMovies(int keywordId, int page) {
        return service.getMovies(keywordId, BuildConfig.TMDB_API_KEY, TmdbConfigKt.en_US, AdultUtil.INSTANCE.includeAdult(Moviemade.appContext), page)
            .subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread());
    }
}

File: app/src/main/java/org/michaelbel/moviemade/ui/modules/keywords/KeywordRepository.kt
package org.michaelbel.moviemade.ui.modules.keywords

import io.reactivex.Observable
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import org.michaelbel.moviemade.BuildConfig
import org.michaelbel.moviemade.Moviemade
import org.michaelbel.moviemade.data.entity.MoviesResponse
import org.michaelbel.moviemade.data.service.KeywordsService
import org.michaelbel.moviemade.utils.AdultUtil
import org.michaelbel.moviemade.utils.LangUtil

class KeywordRepository internal constructor(private val service: KeywordsService) : KeywordContract.Repository {

    override fun getMovies(keywordId: Int, page: Int): Observable<MoviesResponse> {
        return service.getMovies(keywordId, BuildConfig.TMDB_API_KEY, LangUtil.getLanguage(Moviemade.appContext), AdultUtil.includeAdult(Moviemade.appContext), page)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
    }
}